src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -66,768 +67,819 @@ parseAuthorizeCredentials(Json::Value const& jv)
         {
-            expectedType = ltBRIDGE;
+            jvResult[jss::error] = "malformedRequest";
+            return std::nullopt;
+        }
+        return uNodeIndex;
+    }
 
-            // return the keylet for the specified bridge or nullopt if the
-            // request is malformed
-            auto const maybeKeylet = [&]() -> std::optional<Keylet> {
-                try
-                {
-                    if (!context.params.isMember(jss::bridge_account))
-                        return std::nullopt;
-
-                    auto const& jsBridgeAccount =
-                        context.params[jss::bridge_account];
-                    if (!jsBridgeAccount.isString())
-                    {
-                        return std::nullopt;
-                    }
-                    auto const account =
-                        parseBase58<AccountID>(jsBridgeAccount.asString());
-                    if (!account || account->isZero())
-                    {
-                        return std::nullopt;
-                    }
-
-                    // This may throw and is the reason for the `try` block. The
-                    // try block has a larger scope so the `bridge` variable
-                    // doesn't need to be an optional.
-                    STXChainBridge const bridge(context.params[jss::bridge]);
-                    STXChainBridge::ChainType const chainType =
-                        STXChainBridge::srcChain(
-                            account == bridge.lockingChainDoor());
-                    if (account != bridge.door(chainType))
-                        return std::nullopt;
-
-                    return keylet::bridge(bridge, chainType);
-                }
-                catch (...)
-                {
-                    return std::nullopt;
-                }
-            }();
+    if (!claim_id.isObject() ||
+        !(claim_id.isMember(sfIssuingChainDoor.getJsonName()) &&
+          claim_id[sfIssuingChainDoor.getJsonName()].isString()) ||
+        !(claim_id.isMember(sfLockingChainDoor.getJsonName()) &&
+          claim_id[sfLockingChainDoor.getJsonName()].isString()) ||
+        !claim_id.isMember(sfIssuingChainIssue.getJsonName()) ||
+        !claim_id.isMember(sfLockingChainIssue.getJsonName()) ||
+        !claim_id.isMember(jss::xchain_owned_claim_id))
+    {
+        jvResult[jss::error] = "malformedRequest";
+        return std::nullopt;
+    }
 
-            if (maybeKeylet)
-            {
-                uNodeIndex = maybeKeylet->key;
-            }
-            else
-            {
-                uNodeIndex = beast::zero;
-                jvResult[jss::error] = "malformedRequest";
-            }
+    // if not specified with a node id, a claim_id is specified by
+    // four strings defining the bridge (locking_chain_door,
+    // locking_chain_issue, issuing_chain_door, issuing_chain_issue)
+    // and the claim id sequence number.
+    auto lockingChainDoor = parseBase58<AccountID>(
```

Comment:
Could be `const`?

Reply:
Fixed in https://github.com/XRPLF/rippled/pull/5199/commits/3d5ef0fb972cf519f9c890126e2364e5316a014e